src/components/JobCard/JobCard.jsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { useNavigate } from "react-router-dom";
+import {
+  Flex,
+  Box,
+  Heading,
+  Text,
+  Button,
+  IconButton,
+  Stack,
+  Icon,
+  Divider,
+  useMediaQuery,
+} from "@chakra-ui/react";
+import { MdFavoriteBorder, MdFavorite } from "react-icons/md"; // Assuming you're using Material Icons for the save/unsave icon
+import { FaMapMarkerAlt } from "react-icons/fa";
+
+function JobCard(props) {
+  const navigate = useNavigate();
+
+  const handleClick = (id) => {
+    props.setSelectedJob(id);
+  };
+
+  const handleApplyClick = (id) => {
+    props.setSelectedJob(id);
+    navigate(`../apply/${id}`);
+  };
+
+  return (
+    <>
```

Comment:
Same as before, if all of the JSX is wrapped in the `Flex` element, then we might not need the `<></>`